src/wp-includes/class-wp-token-map.php

```diff
@@ -0,0 +1,818 @@
+				break;
+			}
+			$group        = $this->large_words[ $index ];
+			$group_length = strlen( $group );
+			$comment_line = "{$i3}//";
+			$data_line    = "{$i3}\"";
+			$at           = 0;
+			while ( $at < $group_length ) {
+				$token_length   = unpack( 'C', $group[ $at++ ] )[1];
+				$token          = substr( $group, $at, $token_length );
+				$at            += $token_length;
+				$mapping_length = unpack( 'C', $group[ $at++ ] )[1];
+				$mapping        = substr( $group, $at, $mapping_length );
+				$at            += $mapping_length;
+
+				$token_digits   = str_pad( dechex( $token_length ), 2, '0', STR_PAD_LEFT );
+				$mapping_digits = str_pad( dechex( $mapping_length ), 2, '0', STR_PAD_LEFT );
+
+				$mapping = preg_replace_callback(
+					"~[\\x00-\\x1f\\x22\\x5c]~",
+					static function ( $match_result ) {
+						switch ( $match_result[0] ) {
+							case '"':
+								return '\\"';
+
+							case '\\':
+								return '\\\\';
+
+							default:
+								$hex = dechex( ord( $match_result[0] ) );
+								return "\\x{$hex}";
+						}
+					},
+					$mapping
+				);
+
+				$comment_line .= " {$prefix}{$token}[{$mapping}]";
+				$data_line    .= "\\x{$token_digits}{$token}\\x{$mapping_digits}{$mapping}";
+			}
+			$comment_line .= ".\n";
+			$data_line    .= "\",\n";
+
+			$output .= $comment_line;
+			$output .= $data_line;
+		}
+
+		$output .= "{$i2}),\n";
+
+		$small_words  = array();
+		$small_length = strlen( $this->small_words );
+		$at           = 0;
+		while ( $at < $small_length ) {
+			$small_words[] = substr( $this->small_words, $at, $this->key_length + 1 );
+			$at           += $this->key_length + 1;
+		}
+
+		$small_text = str_replace( "\x00", '\x00', implode( '', $small_words ) );
+		$output    .= "{$i2}\"small_words\" => \"{$small_text}\",\n";
+
+		$output .= "{$i2}\"small_mappings\" => array(\n";
+		foreach ( $this->small_mappings as $mapping ) {
+			$output .= "{$i3}\"{$mapping}\",\n";
+		}
+		$output .= "{$i2})\n";
+		$output .= "{$i1})\n";
+		$output .= ')';
+
+		return $output;
+	}
+
+	/**
+	 * Compares two strings, returning the longest, or whichever
+	 * is first alphabetically if they are the same length.
+	 *
+	 * This is an important sort when building the token map because
+	 * it should not form a match on a substring of a longer potential
+	 * match. For example, it should not detect `Cap` when matching
+	 * against the string `CapitalDifferentialD`.
+	 *
+	 * @since 6.6.0
+	 *
+	 * @param string $a First string to compare.
+	 * @param string $b Second string to compare.
+	 * @return int -1 or lower if `$a` is less than `$b`; 1 or greater if `$a` is greater than `$b`, and 0 if they are equal.
+	 */
+	private static function longest_first_then_alphabetical( $a, $b ) {
+		if ( $a === $b ) {
+			return 0;
+		}
+
+		$length_a = strlen( $a );
+		$length_b = strlen( $b );
+
+		// Longer strings are less-than for comparison's sake.
+		if ( $length_a !== $length_b ) {
+			return $length_b - $length_a;
```

Comment:
This is a perfect place for the spaceship operator comparison to return -1, 0, or 1 `return $length_b <=> $length_a;`

Reply:
are we sure about this?

we need to make sure that longer strings sort before shorter ones, so for example, `ba` appears before `a`

```php
-1 === longest_first_then_alphabetical( 'ba', 'a' );
 1 === 'ba' <=> 'a';
 1 === strcmp( 'ba', 'a' );
```

Reply:
Yes, I think this is correct. I meant specifically the length comparison could be `return $length_b <=> $length_a;`. The result is basically the same, but it returns `-1`, `0`, or `1` instead of `<0`, `0`, or `>0`.

<details>

<summary>demo</summary>

```php
<?php

function longest_first_then_alphabetical( $a, $b ) {
	if ( $a === $b ) {
		return 0;
	}


	$length_a = strlen( $a );
	$length_b = strlen( $b );


	// Longer strings are less-than for comparison's sake.
	if ( $length_a !== $length_b ) {
		return $length_b - $length_a;
	}


	return strcmp( $a, $b );
}
	
function longest_first_then_alphabetical_spaceship( $a, $b ) {
	if ( $a === $b ) {
		return 0;
	}


	$length_a = strlen( $a );
	$length_b = strlen( $b );


	// Longer strings are less-than for comparison's sake.
	if ( $length_a !== $length_b ) {
		// 👇👇👇 This is the only different line 👇👇👇
		return $length_b <=> $length_a;
	}


	return strcmp( $a, $b );
}
	
	
var_dump(
    longest_first_then_alphabetical( 'baa', 'a'),
    longest_first_then_alphabetical_spaceship( 'baa', 'a'),
    "\n",

    longest_first_then_alphabetical( 'abb', 'a'),
    longest_first_then_alphabetical_spaceship( 'abb', 'a'),
    "\n",

    longest_first_then_alphabetical( 'a', 'abb'),
    longest_first_then_alphabetical_spaceship( 'a', 'abb'),
    "\n",

    longest_first_then_alphabetical( 'a', 'abb'),
    longest_first_then_alphabetical_spaceship( 'a', 'abb'),
    "\n",

    longest_first_then_alphabetical( 'a', 'baa'),
    longest_first_then_alphabetical_spaceship( 'a', 'baa'),
    "\n",

    longest_first_then_alphabetical( 'a', 'baa'),
    longest_first_then_alphabetical_spaceship( 'a', 'baa'),
    "\n",

    longest_first_then_alphabetical( 'a', 'a'),
    longest_first_then_alphabetical_spaceship( 'a', 'a'),
    "\n",

    longest_first_then_alphabetical( 'b', 'a'),
    longest_first_then_alphabetical_spaceship( 'b', 'a'),
    "\n",

    longest_first_then_alphabetical( 'a', 'b'),
    longest_first_then_alphabetical_spaceship( 'a', 'b'),
    "\n",

);
```
int(-2)
int(-1)
string(1) "
"
int(-2)
int(-1)
string(1) "
"
int(2)
int(1)
string(1) "
"
int(2)
int(1)
string(1) "
"
int(2)
int(1)
string(1) "
"
int(2)
int(1)
string(1) "
"
int(0)
int(0)
string(1) "
"
int(1)
int(1)
string(1) "
"
int(-1)
int(-1)
string(1) "
"
```

</details>